Libplanet/Store/IStateStore.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Immutable;
+using System.Security.Cryptography;
+using Bencodex.Types;
+using Libplanet.Action;
+using Libplanet.Blockchain;
+using Libplanet.Blocks;
+
+namespace Libplanet.Store
+{
+    /// <summary>
+    /// An interface to store states.
+    /// </summary>
+    public interface IStateStore
+    {
+        /// <summary>
+        /// Sets states mapped as relation <see cref="Block{T}.Hash"/> → states.
+        /// It guarantees <see cref="GetState"/> will return the same state if you passed same
+        /// <paramref name="blockHash"/> and haven't written overwrite the value, storage was
+        /// not broken.
+        /// </summary>
+        /// <param name="blockHash">The <see cref="Block{T}.Hash"/> to set states.</param>
+        /// <param name="states">The dictionary has multiple paris, state key → state.</param>
+        /// <param name="blockGetter">The getter returns the block corresponded to
+        /// <see cref="HashDigest{T}"/>–typed parameter.</param>
+        /// <typeparam name="T">An <see cref="IAction"/> type. It should match to
+        /// <paramref name="blockGetter"/>'s <see cref="Block{T}"/>–typed return value's
```

Comment:
```suggestion
        /// <paramref name="blockGetter"/>'s <see cref="Block{T}"/>-typed return value's
```

En dashes shouldn't be used for [compound modifiers][1], but hyphens do instead.

[1]: https://en.wikipedia.org/wiki/Hyphen#Compound_modifiers